### PR TITLE
update cmake setup so that the library can be used by SwiftPM cmake build

### PR DESCRIPTION
**Pull request by tomerd**
motivation: integrate with SwiftPM

changes:
* rename the top level project to SwiftSystem instead of swift-system to align with other projects
* rename System to SystemPackage to align with SwifPM setup
* update architectures to include arm64

### CODE REVIEW

**Comment by tomerd** [comment]
cc @compnerd

**Comment by tomerd** [comment]
@swift-ci please test

**Comment by tomerd** [comment]
cc @compnerd

**Comment by milseman** [comment]
@compnerd what do you think of these changes?

File: cmake/modules/SwiftSystemConfig.cmake.in [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#[[
+This source file is part of the Swift System open source Project
+
+Copyright (c) 2021 Apple Inc. and the Swift System project authors
+Licensed under Apache License v2.0 with Runtime Library Exception
+
+See https://swift.org/LICENSE.txt for license information
+#]]
+
+if(NOT TARGET SwiftSystem)
```
Comment (compnerd) [inline cmake/modules/SwiftSystemConfig.cmake.in:10]: This needs to change to `SystemPackage` since you renamed the target.

File: cmake/modules/SwiftSupport.cmake [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #[[
-This source file is part of the Swift System open source project
+This source file is part of the Swift System open source Project
```
Comment (compnerd) [inline cmake/modules/SwiftSupport.cmake:2]: Nit: unnecessary change

**Review by compnerd — CHANGES_REQUESTED** [review]
I think that this should be fine - I can see this causing some issues for the installers, but we can deal with that subsequently (the manifest needing to be updated).

I'm still unhappy with the change - I don't think that this should be called SystemPackage on anything but macOS - the library name should be libSystem.so or System.dll on non-macOS targets (it has to be different on macOS to avoid the conflict with libSystem).  That said, I don't think that we need to hold this up for dealing with the name.

The change to SwiftSystemConfig.cmake.in though is important - it isn't doing what it is supposed to do.

**Comment by milseman** [comment]
If we wanted to in the future ship System as a binary with toolchains ala corelibs and Darwin/Glibc overlays, would we have the distinction between source and binary distribution on all platforms?

File: cmake/modules/SwiftSupport.cmake (hunk 2 above)
Comment (tomerd) [reply cmake/modules/SwiftSupport.cmake:2]:
```suggestion
This source file is part of the Swift System open source project
```

File: cmake/modules/SwiftSystemConfig.cmake.in (hunk 1 above)
Comment (tomerd) [reply cmake/modules/SwiftSystemConfig.cmake.in:10]:
```suggestion
if(NOT TARGET SystemPackage)
```

**Comment by tomerd** [comment]
> I think that this should be fine - I can see this causing some issues for the installers, but we can deal with that subsequently (the manifest needing to be updated).

:+1:by "installers" you mean Windows installers?

> I'm still unhappy with the change - I don't think that this should be called SystemPackage on anything but macOS - the library name should be libSystem.so or System.dll on non-macOS targets (it has to be different on macOS to avoid the conflict with libSystem). That said, I don't think that we need to hold this up for dealing with the name.

As you probably know, my goal is to integrate SwiftSystem into SwiftPM and given that the toolchain build uses CMake we need to make that work. I have no strong opinion about how to achieve it other that the current CMake setup does not work in this context so needs to be fixed. Do you a different approach you want to propose as an alternative to this PR and the downstream ones in TSC, SwiftP, LSP, etc?

> The change to SwiftSystemConfig.cmake.in though is important - it isn't doing what it is supposed to do.

done, please confirm this is fine now

**Comment by tomerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
> thanks @compnerd
> 
> > I think that this should be fine - I can see this causing some issues for the installers, but we can deal with that subsequently (the manifest needing to be updated).
> 
> 👍
> 
> by "installers" you mean Windows installers?

Yes.  There will be changes needed in swift-installer-scripts to update the manifest.  But that should happen at the same time as updating the tag.

> > I'm still unhappy with the change - I don't think that this should be called SystemPackage on anything but macOS - the library name should be libSystem.so or System.dll on non-macOS targets (it has to be different on macOS to avoid the conflict with libSystem). That said, I don't think that we need to hold this up for dealing with the name.
> 
> As you probably know, my goal is to integrate SwiftSystem into SwiftPM and given that the toolchain build uses CMake we need to make that work. I have no strong opinion about how to achieve it other that the current CMake setup does not work in this context so needs to be fixed. Do you a different approach you want to propose as an alternative to this PR and the downstream ones in TSC, SwiftP, LSP, etc?

Yes, I understand that.  I think that changing the name doesn't need to hold this up for now - this is matching the behaviour in Package.swift.  I'm suggesting that the target name be changed much like it is in Foundation:

```
#if os(iOS) || os(macOS) || os(tvOS) || os(watchOS)
import SwiftSystem
#else
import System
#endif
```

> > The change to SwiftSystemConfig.cmake.in though is important - it isn't doing what it is supposed to do.
> 
> done, please confirm this is fine now

**Review by compnerd — APPROVED** [review]
